accepted/00000-maintenance-windows.md

```diff
@@ -0,0 +1,191 @@
+- Exceptions should be supported (EXDATE)
+- Events for a full day (anniversary) or multiple days must be considered.
+- The Summary may define the schedule name (See also Maintenance Schedule).
+
+### Maintenance Schedule
+
+A Maintenance Schedule is a set of Maintenance Windows. A Maintenance Schedule has a name/label.
+The data should be stored as ICalendar data (https://tools.ietf.org/html/rfc5545).
+In the first iteration importing the data created by an external tool is required.
+Uyuni will not provide a User interface for generating or manipulating Maintenance Windows.
+
+This is a task for future iterations.
+
+To parse the windows we need a library for ICalendar files. E.g. ical4j(https://github.com/ical4j/ical4j).
+
+There could be two types of Schedules for the first iteration
+
+1. One ICalendar file represent one schedule
+
+Every `VEVENT` describe a maintenance window.
+
+2. One ICalendar file contains multiple schedules
+
+Every `VEVENT` must have a `SUMMARY`. All events with the same "summary" belong to the same schedule.
+The summary must match the schedule name in the Uyuni Database.
+
+### Adding a new Maintenance Schedule
+
+In the left sidebar a new page should be added below `Schedule`.
+In this page you can see a list of existing schedules. You can edit them, when you select an existing schedule.
+There should be a "Create Maintenance Schedule" link in the upper left corner which open a new page where the following values can be specified:
+
+- Schedule Name
+- Schedule Type (Single, Multi, ....)
+- Data (ICalendar content) with file upload option
+
+A "Create" button will create the new schedule. The ICalendar data are stored in the database.
+
+
+### Updating a Maintenance Schedule
+
+A Maintenance Schedule will change over the time. A Maintenance Window will be moved or removed on short notice,
+or a new Maintenance Window slot will be created.
+
+This happens while updating the ICalendar data of a maintenance schedule.
+
+While adding a new slot is a non-issue, the movement or removal of a window needs to be handled properly.
+
+As it is nearly impossible to identify a change of a special event we need to follow a different strategy.
+When we refresh or upload a changed ICalendar file, we need to re-evaluate all scheduled actions of systems using the
+new schedule.
+We also need to provide a resolve strategy. This can happen while implementing modules to move actions
+to a different time. Possible modules could be:
+
+- Nearest: Use the nearest maintenance window
+- Previous: Use the maintenance window before the scheduled date
+- Next: Use the maintenance window after the scheduled date
+- Cancel: Cancel the Action
+- Fail: do not import the new schedule at all and keep the existing one
+
+The customer can configure the order:
+
+  resolveOrder=Nearest,Previous,Next,Fail
+
+
+### Schedule an Action
+
+When a system is assigned to a schedule, only schedules withing a maintenance window should be allowed.
```

Comment:
Should be: `...schedule, only actions within a ...`

Reply:
fixed